pypesto/objective/amici/amici.py

```diff
@@ -667,9 +667,11 @@ def check_gradients_match_finite_differences(
         """
         if x is None and "petab_problem" in dir(self.amici_object_builder):
             x = self.amici_object_builder.petab_problem.x_nominal_scaled
-            x_free = self.amici_object_builder.petab_problem.x_free_indices
+            kwargs = {
+                "x_free": self.amici_object_builder.petab_problem.x_free_indices
+            }
```

Comment:
```suggestion
            kwargs["x_free"] = self.amici_object_builder.petab_problem.x_free_indices
```
The current implementation would just ignore any other kwargs.

Reply:
Thanks!